src/openrct2/ride/Vehicle.cpp

```diff
@@ -914,18 +914,20 @@ template<typename T> int32_t Train<T>::Mass()
     return totalMass;
 }
 
-bool Vehicle::SoundCanPlay() const
+using VehicleSoundParamList = FixedVector<OpenRCT2::Audio::VehicleSoundParams, OpenRCT2::Audio::MaxVehicleSounds>;
+
+static bool VehicleSoundCanPlay(Vehicle* vehicle)
```

Comment:
Will we do a follow on PR that moves all the sound updating into a separate class?

Reply:
Sounds like a good idea

Reply:
Perhaps wait until the music PR is merged first as that already moves the ride music out. Unless there should be RideAudio and VehicleAudio.

Reply:
I see, then I'll revert the vehicle part.